File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/studentService.js
import api from "./api";

const BASE_URL = "/students";

const studentService = {

  // Get All Students
  getAllStudents: async () => {

    const response = await api.get(BASE_URL);

    return response.data;

  },

  // Get Student By Id
  getStudentById: async (id) => {

    const response = await api.get(`${BASE_URL}/${id}`);

    return response.data;

  },

  // Create Student
  createStudent: async (student) => {

    const response = await api.post(
      BASE_URL,
      student
    );

    return response.data;

  },

  // Update Student
  updateStudent: async (id, student) => {

    const response = await api.put(
      `${BASE_URL}/${id}`,
      student
    );

    return response.data;

  },

  // Delete Student
  deleteStudent: async (id) => {

    const response = await api.delete(
      `${BASE_URL}/${id}`
    );

    return response.data;

  },

  // Get Student By Roll Number
  getStudentByRollNumber: async (rollNumber) => {

    const response = await api.get(
      `${BASE_URL}/roll/${rollNumber}`
    );

    return response.data;

  },

  // Get Students By Department
  getStudentsByDepartment: async (departmentId) => {

    const response = await api.get(
      `${BASE_URL}/department/${departmentId}`
    );

    return response.data;

  },

  // Get Students By Course
  getStudentsByCourse: async (courseId) => {

    const response = await api.get(
      `${BASE_URL}/course/${courseId}`
    );

    return response.data;

  },

  // Get Students By Semester
  getStudentsBySemester: async (semesterId) => {

    const response = await api.get(
      `${BASE_URL}/semester/${semesterId}`
    );

    return response.data;

  }

};

export default studentService;

File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/StudentService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.student.StudentRequest;
import com.timetable.dto.student.StudentResponse;

public interface StudentService {

    /**
     * Create Student
     */
    StudentResponse saveStudent(StudentRequest request);

    /**
     * Update Student
     */
    StudentResponse updateStudent(Long studentId, StudentRequest request);

    /**
     * Delete Student
     */
    void deleteStudent(Long studentId);

    /**
     * Get Student By ID
     */
    StudentResponse getStudentById(Long studentId);

    /**
     * Get All Students
     */
    List<StudentResponse> getAllStudents();

    /**
     * Get Student By Roll Number
     */
    StudentResponse getStudentByRollNumber(String rollNumber);

    /**
     * Get Students By Department
     */
    List<StudentResponse> getStudentsByDepartment(Long departmentId);

    /**
     * Get Students By Course
     */
    List<StudentResponse> getStudentsByCourse(Long courseId);

    /**
     * Get Students By Semester
     */
    List<StudentResponse> getStudentsBySemester(Long semesterId);
}